render/render.js

```diff
@@ -64,16 +64,21 @@ module.exports = function($window) {
 		insertNode(parent, fragment, nextSibling)
 		return fragment
 	}
-	function createElement(parent, vnode, hooks, ns, nextSibling) {
-		var tag = vnode.tag
-		switch (vnode.tag) {
-			case "svg": ns = "http://www.w3.org/2000/svg"; break
-			case "math": ns = "http://www.w3.org/1998/Math/MathML"; break
+	function getNamespace(vnode) {
+		var ns = {
+			svg: "http://www.w3.org/2000/svg",
+			math: "http://www.w3.org/1998/Math/MathML"
 		}
 
+		return ns = vnode.attrs && vnode.attrs.xmlns || ns[vnode.tag]
```

Comment:
`return ns = vnode.attrs` smells fishy...

Also, you could put the `ns` object in the parent scope to avoid re-creating it every time `getNameSpace()` is called.